lib/src/lib/voicemeeter-connector.ts

```diff
@@ -333,6 +368,185 @@ export default class Voicemeeter {
+            koffi.pointer(koffi.proto("long __stdcall AudioCallback(void* lpUser, long nCommand, void* lpData, long nnn)"))
+        );
+
+        const result = libVM.VBVMR_AudioCallbackRegister(mode, this.registeredAudioCallbackPointers[mode], lpUser ?? null, clientNamePtr);
+        const outClientName = clientNamePtr.toString().replace("/\u0000+$/g", "");
+
+        switch (result) {
+            case 0: {
+                return;
+            }
+            case -1: {
+                throw new Error("Failed to register audio callback");
+            }
+            case -2: {
+                throw new Error(`Audio callback already registered by: ${outClientName}`);
+            }
+            default: {
+                throw new Error(`Unexpected result registering audio callback: ${result}`);
+            }
+        }
+    };
+
+    public startAudioCallback = (): void => {
+        const result = libVM.VBVMR_AudioCallbackStart();
+        switch (result) {
+            case 0: {
+                return;
+            }
+            case -1: {
+                throw new Error("Failed to start audio callback");
+            }
+            case -2: {
+                throw new Error("No audio callback registered");
+            }
+            default: {
+                throw new Error(`Unexpected result starting audio callback ${result}`);
+            }
+        }
+    };
+
+    public stopAudioCallback = (): void => {
+        const result = libVM.VBVMR_AudioCallbackStop();
+        switch (result) {
+            case 0: {
+                return;
+            }
+            case -1: {
+                throw new Error("Failed to stop audio callback");
+            }
+            case -2: {
+                throw new Error("No audio callback registered");
+            }
+            default: {
+                throw new Error(`Unexpected result stopping audio callback: ${result}`);
+            }
+        }
+    };
+
+    public unregisterAudioCallback = (mode: AudioCallbackModes): void => {
+        if (this.registeredAudioCallbackPointers[mode] === undefined) {
+            throw new Error(`No audio callback registered for "${mode}" in library`);
+        }
+
+        const result = libVM.VBVMR_AudioCallbackUnregister(this.registeredAudioCallbackPointers[mode]);
+        switch (result) {
+            case 0: {
+                if (this.audioCallbackEnded) {
+                    koffi.unregister(this.registeredAudioCallbackPointers[mode]);
+                    delete this.registeredAudioCallbackPointers[mode];
+                } else {
+                    this.pendingAudioCallbackUnregisters[mode] = true;
+                }
+                return;
+            }
+            case -1: {
+                throw new Error("Failed to unregister audio callback");
+            }
+            case -2: {
+                delete this.registeredAudioCallbackPointers[mode];
+                throw new Error("Callback already unregistered");

```

Comment:
`case -2` shouldn't happen with how the library deals with it. I'm also not really sure if this is worth throwing an error for?